blenderproc/python/types/URDFUtility.py

```diff
@@ -0,0 +1,372 @@
+from typing import Union, List
+import numpy as np
+from mathutils import Vector, Euler, Color, Matrix, Quaternion
+
+import bpy
+
+from blenderproc.python.utility.Utility import Utility
+from blenderproc.python.types.EntityUtility import Entity
+from blenderproc.python.types.MeshObjectUtility import MeshObject
+from blenderproc.python.types.BoneUtility import get_constraint, set_ik_constraint, \
+    set_ik_limits_from_rotation_constraint
+
+
+class URDFObject(Entity):
+    def __init__(self, armature: bpy.types.Armature, links: List[Link], xml_tree: Union["urdfpy.URDF", None] = None):
+        super().__init__(bpy_object=armature)
+
+        object.__setattr__(self, "links", links)
+        object.__setattr__(self, "xml_tree", xml_tree)
+        object.__setattr__(self, "ik_bone_constraint", None)
+        object.__setattr__(self, "ik_bone_controller", None)
+        object.__setattr__(self, "fk_ik_mode", None)
+        object.__setattr__(self, "ik_link", None)
+        object.__setattr__(self, 'ik_bone_offset', None)
+
+    def get_all_urdf_objs(self) -> List[Union[Link, Inertial, MeshObject]]:
+        """ Returns a list of all urdf-related objects.
+
+        :return: List of all urdf-related objects.
+        """
+        objs = []
+        for link in self.links:
+            objs.append(link)
+            objs.extend(link.get_all_objs())
+        return objs
+
+    def get_all_collision_objs(self) -> List[MeshObject]:
+        """ Returns a list of all collision objects.
+
+        :return: List of all collision objects.
+        """
+        return [obj for obj in self.get_all_urdf_objs() if 'collision' in obj.get_name()]
+
+    def get_all_inertial_objs(self) -> List[Inertial]:
+        """ Returns a list of all inertial objects.
+
+        :return: List of all inertial objects.
+        """
+        return [obj for obj in self.get_all_urdf_objs() if isinstance(obj, Inertial)]
+
+    def get_all_visual_objs(self) -> List[MeshObject]:
+        """ Returns a list of all visual objects.
+
+        :return: List of all visual objects.
+        """
+        return [obj for obj in self.get_all_urdf_objs() if 'visual' in obj.get_name()]
+
+    def hide_irrelevant_objs(self):
```

Comment:
This name is a bit confusing? 

`hide_collision_and_inertial_objs` is only a bit longer and much more descriptive.

Reply:
You're right; changed it to `hide_link_and_collision_inertial_objs` since it hides the links and their collision / inertial objects